consensus/qbft/src/integration-test/resources/genesis_validator_contract_shanghai.json

```diff
@@ -0,0 +1,46 @@
+{
+	"nonce": "0x0",
+	"timestamp": "0x100",
+	"extraData": "0xe5a00000000000000000000000000000000000000000000000000000000000000000c0c080c0",
+	"gasLimit": "0x29b92700",
+	"difficulty": "0x1",
+	"mixHash": "0x63746963616c2062797a616e74696e65206661756c7420746f6c6572616e6365",
+	"coinbase": "0x0000000000000000000000000000000000000000",
+	"alloc": {
+		"64d9be4177f418bcf4e56adad85f33e3a64efe22": {
+			"balance": "0x446c3b15f9926687d2c40534fdb564000000000000"
+		},
+		"9f66f8a0f0a6537e4a36aa1799673ea7ae97a166": {
+			"balance": "0x446c3b15f9926687d2c40534fdb564000000000000"
+		},
+		"a7f25969fb6f3d5ac09a88862c90b5ff664557a7": {
+			"balance": "0x446c3b15f9926687d2c40534fdb564000000000000"
+		},
+		"f4bbfd32c11c9d63e9b4c77bb225810f840342df": {
+			"balance": "0x446c3b15f9926687d2c40534fdb564000000000000"
+		},
+		"0x0000000000000000000000000000000000008888": {
+			"comment": "validator smart contract. This is compiled from validator_contract.sol using solc --evm-version byzantium --bin-runtime validator_contract.sol",
+			"balance": "0",
+			"code": "608060405234801561001057600080fd5b5060043610610048576000357c010000000000000000000000000000000000000000000000000000000090048063b7ab4db51461004d575b600080fd5b61005561006b565b604051610062919061017e565b60405180910390f35b606060008054806020026020016040519081016040528092919081815260200182805480156100ef57602002820191906000526020600020905b8160009054906101000a900473ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff16815260200190600101908083116100a5575b5050505050905090565b60006101058383610111565b60208301905092915050565b61011a816101d9565b82525050565b600061012b826101b0565b61013581856101c8565b9350610140836101a0565b8060005b8381101561017157815161015888826100f9565b9750610163836101bb565b925050600181019050610144565b5085935050505092915050565b600060208201905081810360008301526101988184610120565b905092915050565b6000819050602082019050919050565b600081519050919050565b6000602082019050919050565b600082825260208201905092915050565b60006101e4826101eb565b9050919050565b600073ffffffffffffffffffffffffffffffffffffffff8216905091905056fea26469706673582212206d880cf012c1677c691bf6f2f0a0e4eadf57866ffe5cd2d9833d3cfdf27b15f664736f6c63430008060033",
```

Comment:
Would give more confidence if we used a Shanghai compiled contract in the Shanghai tests then we would know it's using the Shanghai EVM opcodes

Reply:
Agreed. I've recompiled with `shanghai`.